packages/atomic/src/components/facets/atomic-category-facet/atomic-category-facet.tsx

```diff
@@ -156,6 +160,18 @@ export class AtomicCategoryFacet
   @Prop() public injectionDepth = 1000;
   // @Prop() public customSort?: string; TODO: KIT-753 Add customSort option for facet
 
+  @FocusTarget()
+  protected showMoreFocus!: FocusTargetController;
+
+  @FocusTarget()
+  protected showLessFocus!: FocusTargetController;
```

Comment:
This is a good example of what `@FocusTarget` does.

When we press "show more", it may or may not be removed. If it is removed, then we want to focus on the "show less" button. Otherwise, we want to leave the focus on the "show more" button. Additionally, these changes will occur only after a search has been completed, which may be multiple `componentDidRender` away.

The decorator solves that issue by diving the whole process into three methods:
* setTarget
* focusAfterSearch
* disableForCurrentSearch

Calling `focusAfterSearch` activates a flag that indicates that we want to focus on the target as soon as:
* It's available
* A new search was performed

Calling `setTarget` will set the target to be focused.

Calling `disableForCurrentSearch` indicates that we shouldn't focus on an element during the current search. In most cases, calling this function isn't necessary, but it's useful when we can't predict whether an element should be focused purely based on whether it's rendered or not.

Reply:
> Why opt for a decorator vs. a class or builder function? e.g.

Because we override some lifecycle methods. We can do it from a class or builder too, but I felt like that kind of behaviour is only expected from a decorator.

> Why use `protected` properties instead of `private` ones?

I don't remember why. I remember that I used to have a minor issue with the code editor that made `protected` more convenient, but switching it back to `private` right now makes no difference. I'll switch them to `private`.